demos/workbench/chat_app.py

```diff
@@ -331,11 +331,12 @@ def update_semantic_cache(self, use_semantic_cache: bool):
         if self.use_semantic_cache and self.index_name:
             self.llmcache = self.make_semantic_cache()
             self.update_llm()
-
-            if self.vector_store:
-                self.chain = self.build_chain(self.vector_store)
         else:
             self.llmcache = None
+            self.update_llm()
+
+        if self.vector_store:
```

Comment:
This is happening in `self.update_llm()` right?

Reply:
Oh yeah, it is, don't know why it was in here too.

Reply:
Updated